src/components/MiniSearchUI.js

```diff
@@ -7,28 +7,62 @@ import NavbarSearch from '../components/NavbarSearch';
 import SearchResults from '../components/SearchResults';
 import Scrollable from '../components/Scrollable';
 import SearchResultsCount from '../components/SearchResultsCount';
+import ObjectUtils from '../util/ObjectUtils';
+import QueryResponse from '../api/QueryResponse';
 
 type MiniSearchUIProps = {
   /**
    * A scale (1.0 = 100%) to use when rendering the search results part of
    * the MiniSearchUI. Optional—defaults to 100%.
    */
   scale: number;
+  /**
+   * Optional callback to handle searching if not using the context's
+   * Searcher component to do the searching (otherwise the searcher's onSearch method is used).
+   */
+  onSearch: null | (q: string) => void;
```

Comment:
Indicate this is optional: `onSearch?: (q: string) => void;`

The difference between 
A: `onSearch: null | (q: string) => void;` and
B: `onSearch?: (q: string) => void;`

In example A, you are specifying that the user must explicitly pass a prop named `onSearch` and it may have the value of `null` or it could be a function like `(q: string) => void`. Omitting a function is not the same as `null`. If `onSearch` was not passed as a prop, it would be `undefined`. It's easier to quickly see that something is optional using `?` - that's it's specific intention.

Reply:
Done. Made `onSearch`, `updateQuery`, `response`, and `error` optional props. I also removed the `null` option for `onSearch` and `updateQuery` and removed these props from the default props, since these should either be valid functions or not exist at all as props.